src/components/triggersAndActions.ts

```diff
@@ -78,55 +89,100 @@ export const loadScheduledActions = async function (
+    }
   }
 
-  // Evaluate parameters for each Action
-  for (const action of actions) {
-    for (const key in action.parameter_queries) {
-      action.parameter_queries[key] = await evaluateExpression(action.parameter_queries[key], {
-        objects: [payload],
-        pgConnection: PostgresDB,
-      })
-    }
-    // TODO - Error handling
-    // Write each Action with parameters to Action_Queue
-    await PostgresDB.addActionQueue({
-      trigger_event: payload.id,
+  for (const action of [...actionsAsync, ...actionsSequential]) {
+    // Add all actions to Action Queue
+    // TODO - better error handling
+    await DBConnect.addActionQueue({
+      trigger_event: trigger_id,
+      template_id: templateId,
+      sequence: action.sequence,
       action_code: action.code,
-      parameters: action.parameter_queries,
-      status: 'Queued',
+      trigger_payload: payload,
+      parameter_queries: action.parameter_queries,
+      parameters_evaluated: {},
+      status: action.sequence ? 'Processing' : 'Queued',
     })
   }
+  await DBConnect.updateTriggerQueueStatus({ status: 'Actions Dispatched', id: trigger_id })
+
+  // Get sequential Actions from database
+  const actionsToExecute = await DBConnect.getActionsProcessing(templateId)
+
+  // Execute Actions one by one
+  for (const action of actionsToExecute) {
+    const actionPayload = {
+      id: action.id,
+      code: action.action_code,
+      trigger_payload: action.trigger_payload,
+      parameter_queries: action.parameter_queries,
+    }
+    await executeAction(actionPayload, actionLibrary)
+  }
 
-  // Update trigger queue item with success/failure (and log)
-  // If SUCCESS -- Not sure best way to test for this:
-  await PostgresDB.updateTriggerQueue({ status: 'Action Dispatched', id: payload.id })
+  // After all done, set Trigger on table back to NULL
+  DBConnect.resetTrigger(table, record_id)
+}
+
+async function evaluateParameters(
+  parameterQueries: BasicObject,
+  evaluatorParameters: IParameters = {}
+) {
+  const parametersEvaluated: BasicObject = {}
+  try {
+    for (const key in parameterQueries) {
+      parametersEvaluated[key] = await evaluateExpression(
+        parameterQueries[key],
+        evaluatorParameters
+      )
+    }
+    return parametersEvaluated
+  } catch (err) {
+    throw err
+  }
 }
 
 export async function executeAction(
   payload: ActionPayload,
   actionLibrary: ActionLibrary
 ): Promise<boolean> {
+  const evaluatorParams = {
+    objects: [payload.trigger_payload],
+    pgConnection: DBConnect, // Add graphQLConnection, Fetch (API) here when required
+  }
+  // Evaluate parameters
+  const parametersEvaluated = await evaluateParameters(payload.parameter_queries, evaluatorParams)
+
   // TO-DO: If Scheduled, create a Job instead
-  const actionResult = await actionLibrary[payload.code](payload.parameters, PostgresDB)
+  const actionResult = await actionLibrary[payload.code](parametersEvaluated, DBConnect)
```

Comment:
is it possible that actions will use evaluator expression ? If so they would need GraphQL ad Postgres instance injected. I guess we can store them in dbConnectInstance class

Reply:
Actions shouldn't need the evaluator. All parameters are evaluated before being passed in, and there's no need for it within the plugin as its only needed where expressions need to be user-facing.